### PR TITLE
Update CodeQL workflow

### PR DESCRIPTION
**Pull request by XhmikosR**
* update to the latest action tags
* specify `security-and-quality` and `category`

### CODE REVIEW

**Comment by ghost** [comment]
A quick fix for the code's safe problem, hope it helps.
Any suggestion would be appreciated.

**Comment by ghost** [comment]
@XhmikosR：Maybe you can fix the conflict.
@nodejs/nodejs-dev : Any one would be kind to review XhmikosR's submit, as well as mine at https://github.com/nodejs/nodejs.org/pull/4931

**Comment by XhmikosR** [comment]
I'd wait for #4931 after this PR has landed.

**Comment by ovflowd** [comment]
Sorry for the delay, I didn't notice this existed. #4931 got merged. Feel free to rebase.

**Comment by ghost** [comment]
@XhmikosR & @ovflowd:Finally got finished! Too tired to change the code totally without using "fs.existsSync(....)" to meet the safety of the codeQL 

XD

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -4,9 +4,11 @@ on:
   push:
     branches:
       - main
+      - "!dependabot/**"
```
Comment (nschonni) [inline .github/workflows/codeql.yml]: This isn't needed when there is only a allow list of a single branch already

File: scripts/release-post.js [hunk 2]
```diff
@@ -211,18 +211,29 @@ function writeToFile(results) {
   );
 
   return new Promise((resolve, reject) => {
-    if (fs.existsSync(filepath) && process.argv[3] !== '--force') {
```
Comment (nschonni) [inline scripts/release-post.js:214]: This script is used (maybe not anymore?) by the releasers to create their blog posts. I don't think these changes are required. Also not clear why the execute bit is being removed

Comment (richardlau) [reply scripts/release-post.js:214]: > This script is used (maybe not anymore?) by the releasers to create their blog posts.

This script is still used by @nodejs/releasers. It's referenced in https://github.com/nodejs/node/blob/main/doc/contributing/releases.md#17-create-a-blog-post.

Comment (ghost) [reply scripts/release-post.js:214]: > not clear why the execute bit is being removed

I'm not sure whether you mean if I've removed:

```js
if (fs.existsSync(filepath) && process.argv[3] !== '--force') {
```

If so, in fact according to codeQL's suggestion, I used "openSync" to check whether the file is "readable" or not (whether the file exists or not), and if yes without "--force", the file cannot be overridden.

PS: To be honest, it's not easy to process with 'fs.openSync' totally, but now everthing goes fine after my tests locally.

Comment (ghost) [reply scripts/release-post.js:214]: > not clear why the execute bit is being removed

I'm not sure whether you mean if I've removed:

```js
if (fs.existsSync(filepath) && process.argv[3] !== '--force') {
```

If so, in fact according to codeQL's suggestion, I used "openSync" to check whether the file is "readable" or not (whether the file exists or not), and if yes without "--force", the file cannot be overridden.

PS: To be honest, it's not easy (maybe hard to understand) or to process with 'fs.openSync' totally, but now everthing goes fine after my tests locally.

**Comment by ovflowd** [comment]
Hey @MaledongGit any updates here?

**Comment by ovflowd** [comment]
Closing as we're using the built-in CodeQL workflow now.

**Pull request closed without merging**